src/client/components/GraphScene.tsx

```diff
@@ -126,21 +131,54 @@ export const dataSelecter = (rawData: any, provider: string): any => {
   }
 }
 
-export default function GraphScene(): JSX.Element {
-  const [data] = useState([])
+export default function GraphScene(props: GraphSceneProps): JSX.Element {
+  const [graphData, setGraphData] = useState([] as any)
+  const location = useLocation<Location>()
+
+  const getData = async (start: string, end: string): Promise<number> => {
+    let data
+    console.time('getData')
+    props.setLoading(true)
+    let response
+    const uri = `${window.location.origin}/v1/getDataAllPartners/?firstDate=${start}&secondDate=${end}`
```

Comment:
All of this can probably be moved into the `try` block. Then instead of `let data` and `let response` outside of it, you can do `const response = await fetch(uri)` and `const data = await response.json()`.